dhall/src/Dhall/Substitution.hs

```diff
@@ -0,0 +1,21 @@
+module Dhall.Substitution where
+
+import Dhall.Core (subst)
+import Dhall.Syntax (Expr, Var)
+
+import qualified Data.Map
+
+{- | Substitutions map variables to arbitrary Dhall expressions.
+-}
+type Substitutions s a = Data.Map.Map Var (Expr s a)
```

Comment:
Why is the key type `Var` instead of just `Text`?

(Maybe we should add a `Label` newtype for valid labels/identifiers and define 

```haskell
data Var = V Label !Int
```
)

Reply:
The key should definitely be `Text` rather than `Var`.  The index is implicit from the ordering in the `Map`

Reply:
Done